docs/changelogs/v4.0.0.md

```diff
@@ -0,0 +1,559 @@
+
+#2375 Tail fin aerodynamics: address changes requested in pull request #1874 (@abhineet-gupta)
+
+
+#### OLAF 
+
+#1791 Apply the FakeGroundEffect to all FWspan points (@rcorniglion)
+
+#2349 [OLAF] Bug Fix: Twist Not Computed in OLAF cases when Polar Projection is used, resulting in wrong Fn, Ft outputs (@ebranlard)
+
+#2581 OLAF out of bounds (@andrew-platt)
+
+
+#### AeroDyn Driver / AeroDyn\_Inflow\_C\_Bindings interface
+
+#1784 Multi-rotor support in ADI c-bindings interface (@andrew-platt)
+
+#2108 Refactored Aerodyn inflow C binding interface to get blade resolved mesh information (@faisal-bhuiyan)
+
+#2140 ADI: disable OMP in `AeroDyn_Inflow.f90` (@andrew-platt)
+
+#2287 Bug Fix: AeroDyn driver output files contained wrong channel names (@bjonkman)
+
+#2370 Correct frequency error in AeroDyn driver prescribed motions (@hkross)
+
+#2457 Bug: AeroDyn-Inflow WriteOutput Value Ordering (@deslaughter)
+
+#2470 Add blade distributed load output to AeroDyn-InflowWind C bindings (@deslaughter)
+
+#2571 Add new routine and variables to init in the `ADI_c_binding` interface (@andrew-platt)
+
+
+
+### ElastoDyn
+_ElastoDyn_ includes a new yaw friction model, new output channels, several bugfixes, and code cleanup.
+* Input file changes
+
+#2017 Yaw friction implementation (@abhineet-gupta, @kevo331, @rdamiani)
+
+#1760 Add output channels of linear acceleration relative to g in ElastoDyn
+
+#1426 Remove references to ADAMS coupling (@hkross)
+
+#2247 ED: Add inputs for the off-diagonal terms of the platform moment of inertia matrix (@luwang00)
+
+#2561 ED: Extended yaw-friction modeling (@luwang00)
+
+
+
+### ExtInflow (formerly OpenFOAM)
+_ExtInflow_ (also called _ExternalInflow_) is the renamed _OpenFOAM_ module.  The name change better reflects what this module does as it can couple to more than just the _OpenFOAM_ solver used in _SOWFA_.
+
+#1687 ExternalInflow: rename files, change names in files (@andrew-platt)
+
+#2443 Fixes in ExtInflw (heap-buffer-overlow, index flip, empty structure) (@marchdf)
+
+
+
+### ExtLoads
+_ExtLoads_ is a new module for blade resolved fluid-structure interaction with CFD codes such as _AMR-Wind_.  This module is accessed through the CPP interface to OpenFAST.
+
+#1932, #1946 Blade resolved loads from CFD (@gantech, @deslaughter, @psakievich, @jrood-nrel, @andrew-platt)
+
+#2001 ExtLoads: move integers from inputs to parameters (@andrew-platt)
+
+#2009 ExtLoads module: Use pointers for wind (@andrew-platt)
+
+#2226 Fix turbine indexing on ExtLoads Restart (@ndevelder)
+
+#2412 ExtLd: require use of InflowWind (@andrew-platt)
+
+
+### HydroDyn
+The _HydroDyn_ module was completely restructured with wave dynamics split out into a new module, _SeaState_ (documented below).  _HydroDyn_ now accesses wave-field data through a pointer to the data held in the _SeaState_ module.  Other improvements include large yaw angle dynamics, bug fixes, and feature enhancements.
```

Comment:
Add: "wave loads at the displaced position in surge/sway".
Add: "MacCamy-Fuchs diffraction for strip-theory members".
Add: "wave stretching (various models) with load smoothing".
Add: "constrained New Wave theory".
Add: "new strip-theory hydrostatic solution valid for near-horizontal members at the free surface".